public/test/scripts/stacking.js

```diff
@@ -5,58 +5,87 @@ Test.describe(
 
     // elements with visibility: hidden are ignored
     target = await Test.getMouseTarget(140, 140);
-    Test.assert(quadsAreEqual(target._rect, [110, 110, 150, 150]));
+    Test.assert(quadsAreEqual(target._rects, [[110, 110, 150, 150]]));
 
     // elements with pointer-events: none are ignored
     target = await Test.getMouseTarget(240, 140);
-    Test.assert(quadsAreEqual(target._rect, [210, 110, 250, 150]));
+    Test.assert(quadsAreEqual(target._rects, [[210, 110, 250, 150]]));
 
     // the z-index is ignored for unpositioned elements
     target = await Test.getMouseTarget(340, 140);
-    Test.assert(quadsAreEqual(target._rect, [330, 130, 370, 170]));
+    Test.assert(quadsAreEqual(target._rects, [[330, 130, 370, 170]]));
 
     // ... except if they are flex items
     target = await Test.getMouseTarget(40, 240);
-    Test.assert(quadsAreEqual(target._rect, [5, 205, 45, 245]));
+    Test.assert(quadsAreEqual(target._rects, [[5, 205, 45, 245]]));
 
     // ... or grid items
     target = await Test.getMouseTarget(140, 240);
-    Test.assert(quadsAreEqual(target._rect, [105, 205, 145, 245]));
+    Test.assert(quadsAreEqual(target._rects, [[105, 205, 145, 245]]));
 
     // positioned elements without z-index don't create a stacking context
     target = await Test.getMouseTarget(240, 240);
-    Test.assert(quadsAreEqual(target._rect, [210, 210, 250, 250]));
+    Test.assert(quadsAreEqual(target._rects, [[210, 210, 250, 250]]));
 
     // children of positioned elements are in front of children of unpositioned ones
     target = await Test.getMouseTarget(340, 240);
-    Test.assert(quadsAreEqual(target._rect, [310, 210, 350, 250]));
+    Test.assert(quadsAreEqual(target._rects, [[310, 210, 350, 250]]));
+
+    // elements with multiple client rects:
+    // all client rects are highlighted
+    target = await Test.getMouseTarget(60, 340);
+    Test.assert(target._rects.length === 2);
+    // clicking inside the element's bounding client rect but outside all of its
+    // client rects will not select the element
+    target = await Test.getMouseTarget(60, 320);
+    Test.assert(quadsAreEqual(target._rects, [[0, 300, 80, 380]]));
+
+    // unpositioned children of elements with overflow will be clipped
+    target = await Test.getMouseTarget(140, 340);
+    Test.assert(quadsAreEqual(target._rects, [[100, 300, 180, 380]]));
+    // ... but the highlighter will show their unclipped size
+    target = await Test.getMouseTarget(140, 320);
+    Test.assert(quadsAreEqual(target._rects, [[105, 305, 145, 345]]));
+
+    // absolutely positioned elements are not clipped by their unpositioned parent
+    target = await Test.getMouseTarget(240, 340);
+    Test.assert(quadsAreEqual(target._rects, [[210, 310, 250, 350]]));
+
+    // ... but relatively positioned ones are
+    target = await Test.getMouseTarget(340, 340);
+    Test.assert(quadsAreEqual(target._rects, [[300, 300, 380, 380]]));
   }
 );
 
 function quadsAreEqual(q1, q2) {
```

Comment:
Probably could also do something `JSON.stringify(q1) === JSON.stringify(q2)` (that preserves nesting and all, unlike `.toString()` e.g.), but that's just a sugggestion.